From 960516e44dcc494088d1cde1d36f0dd7442739ce Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Mon, 4 May 2020 00:16:00 +0200
Subject: [PATCH 25/98] ARM: dts: meson8b: odroidc1: Enable the Micro USB OTG
 connector

Enable &usb0 which is routed to the Micro USB connector. The port
supports OTG modes and the role switch is implemented by reading out the
"ID" signal from &usb0_phy.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

[dts: rebased onto 6.6.144 for 6.6-lts - trailing offset only, same content]
---
diff --git a/arch/arm/boot/dts/amlogic/meson8b-odroidc1.dts b/arch/arm/boot/dts/amlogic/meson8b-odroidc1.dts
index 941682844..4b311f63d 100644
--- a/arch/arm/boot/dts/amlogic/meson8b-odroidc1.dts
+++ b/arch/arm/boot/dts/amlogic/meson8b-odroidc1.dts
@@ -93,6 +93,20 @@ rtc32k_xtal: rtc32k-xtal-clk {
 		#clock-cells = <0>;
 	};
 
+	usb0_vbus: regulator-usb0-vbus {
+		/* Richtek RT9715EGB */
+		compatible = "regulator-fixed";
+
+		regulator-name = "USB0_VBUS";
+		regulator-min-microvolt = <5000000>;
+		regulator-max-microvolt = <5000000>;
+
+		vin-supply = <&p5v0>;
+
+		gpio = <&gpio_ao GPIOAO_5 GPIO_ACTIVE_HIGH>;
+		enable-active-high;
+	};
+
 	vcc_1v8: regulator-vcc-1v8 {
 		/*
 		 * RICHTEK RT9179 configured for a fixed output voltage of
@@ -363,8 +377,18 @@ &uart_AO {
 	pinctrl-names = "default";
 };
 
-&usb1_phy {
+&usb0 {
 	status = "okay";
+
+	dr_mode = "otg";
+	usb-role-switch;
+	role-switch-default-mode = "peripheral";
+
+	port {
+		usb_to_musbcon: endpoint {
+			remote-endpoint = <&musbcon_to_usb>;
+		};
+	};
 };
 
 &usb1 {
@@ -381,3 +405,24 @@ hub@1 {
 		reset-gpio = <&gpio_ao GPIOAO_4 GPIO_ACTIVE_LOW>;
 	};
 };
+
+&usb0_phy {
+	status = "okay";
+
+	connector {
+		compatible = "gpio-usb-b-connector", "usb-b-connector";
+		type = "micro";
+		id-gpios = <&usb0_phy 0 GPIO_ACTIVE_HIGH>;
+		vbus-supply = <&usb0_vbus>;
+
+		port {
+			musbcon_to_usb: endpoint {
+				remote-endpoint = <&usb_to_musbcon>;
+			};
+		};
+	};
+};
+
+&usb1_phy {
+	status = "okay";
+};
